### PR TITLE
Reject weird archive names, for #1166

### PR DESCRIPTION
**Pull request by Abogical**
This prevents the creation of new archives to have this 'crap'. However this doesn't do anything for old archives which have those weird names.

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
I have not yet made up my mind about whether I want this change, but I'll provide some feedback about the implementation nevertheless.

File: src/borg/helpers.py [hunk 1]
```diff
@@ -728,16 +728,16 @@ class Location:
...
+                         r'(?P<path>[^:]+)(?:::(?P<archive>[^/\s]([^/\s]| )*[^/\s]))?$')
     scp_re = re.compile(r'((?:(?P<user>[^@]+)@)?(?P<host>[^:/]+):)?'
-                        r'(?P<path>[^:]+)(?:::(?P<archive>[^/]+))?$')
+                        r'(?P<path>[^:]+)(?:::(?P<archive>[^/\s]([^/\s]| )*[^/\s]))?$')
     # get the repo from BORG_RE env and the optional archive from param.
     # if the syntax requires giving REPOSITORY (see "borg mount"),
     # use "::" to let it use the env var.
     # if REPOSITORY argument is optional, it'll automatically use the env.
-    env_re = re.compile(r'(?:::(?P<archive>[^/]+)?)?$')
+    env_re = re.compile(r'(?:::(?P<archive>[^/\s]([^/\s]| )*[^/\s])?)?$')
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: This is not just related to you change, but more a general thing:
this code / these regexes is somehow way to complex already (and it's not getting better by making the regexes even more complex).
So, I'ld rather like to simplify / deduplicate it somehow.

File: src/borg/helpers.py [hunk 2]
```diff
@@ -773,6 +773,14 @@ def parse(self, text):
...
         self.archive = m.group('archive')
+        return self._is_archive_name_printable()
+
+    # lookout for non-printable characters
+    def _is_archive_name_printable(self):
+        if self.archive:
+            for char in self.archive:
+                if unicodedata.category(char)[0] == 'C':
+                    return False
         return True
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: this shouldn't be a method, but rather a global utility function.

implementation could looks somehow like:

def printable(s):
    return all(unicodedata.category(c)[0] == 'C' for c in s)

wondering if such a function does not already exist? in stdlib?

File: src/borg/testsuite/helpers.py [hunk 3]
```diff
@@ -104,6 +104,21 @@ def test_no_slashes(self, monkeypatch):
         with pytest.raises(ValueError):
             Location('/some/path/to/repo::archive_name_with/slashes/is_invalid')
 
+    def test_only_printable_characters(self, monkeypatch):
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py]: the utility function then could be tested on its own, disconnected from the archivename context.

File: src/borg/helpers.py (hunk 2 above)
Comment (Abogical) [inline src/borg/helpers.py]: I've looked it up, such a function, dissapointedly, doesn't exist. The closest functions would be string.printable and curses.ascii.isprint(), but they only only on ASCII, not unicode.

Comment (Abogical) [inline src/borg/helpers.py]: Ok I may be wrong, I think the function would be string.isprintable()

**Comment by ThomasWaldmann** [comment]
we had a short discussion on irc about this and feel that we rather won't take this.

the downside of making the regexes even more complex (and that there is always the option to just not feed crappy archive names into borg) somehow outweighs the good sides of this PR.

also, your other "formatting archive names" PR somehow can solve any such issues in another way.

thanks for engaging anyway. :)

**Comment by Abogical** [comment]
Your welcome :)

**Pull request closed without merging**